gen_world_ca.py

```diff
@@ -168,6 +178,9 @@ def biggest_right_region(self):
     return max_region
 
   def regions_connected(self, regionA, regionB):
+    if len(regionB) != len(regionA):
+      # Sometimes one of the regions can be empty
+      return False
```

Comment:
I noticed when running this script in Python 3 sometimes `regionB` is an empty list `[]`. I don't know why this is, but it causes an `IndexError` exception to be raised when trying to compare `regionB[r][c]` below.